### PR TITLE
style: unbreak bindgen on BSDs and Solaris

### PR DESCRIPTION
**Pull request by jbeich**
---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [x] These changes fix Firefox [bug 1339033](https://bugzilla.mozilla.org/show_bug.cgi?id=1339033)

- [x] These changes do not require tests because Tier3 platforms support is incomplete

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15529)

### CODE REVIEW

**Comment by emilio** [comment]
@bors-servo r+

Thanks :)

**Comment by SimonSapin** [comment]
Are these defines specific to Mozilla code or could they be useful to any project using bindgen?

**Comment by jbeich** [comment]
In Gecko `OS_*` defines are only used by `ipc/chromium`, `media/webrtc`, `security/sandbox/chromium` and whatever depends on those bits of Google's code. Other projects generally check `__sun`, `__DragonFly__`, `__FreeBSD__`, `__NetBSD__`, `__OpenBSD__` directly or wrap them in their own flavor (e.g. Qt uses `Q_OS_*`, WebKit uses `WTF_OS_*`).

File: components/style/build_gecko.rs [hunk 1]
```diff
@@ -136,6 +136,16 @@ mod bindings {
             }
             if cfg!(target_os = "linux") {
                 builder = builder.clang_arg("-DOS_LINUX=1");
+            } else if cfg!(target_os = "solaris") {
+                builder = builder.clang_arg("-DOS_SOLARIS=1")
+            } else if cfg!(target_os = "dragonfly") {
+                builder = builder.clang_arg("-DOS_BSD=1").clang_arg("-DOS_DRAGONFLY=1")
+            } else if cfg!(target_os = "freebsd") {
+                builder = builder.clang_arg("-DOS_BSD=1").clang_arg("-DOS_FREEBSD=1")
```
Comment (jbeich) [inline components/style/build_gecko.rs]: Terminating semicolons are missing but `cargo build` in Gecko didn't complain. I guess, it only affects style here.

**Pull request merged**